### ISSUE TITLE
Support for Truncated Backpropagation Through Time with tf.data.TFRecordDataset API

### ISSUE THREAD

**Issue by ghost** [tensorflow/tensorflow#23154]
I am aware that there is support on Tensorflow for truncated backpropagation through time with the tf.contrib.training.batch_sequences_with_states method and the state saving RNN. However this requires the conventional method using the queue runners. I am using the new tf.data.TFRecordDataset API for my my work which does not require the developer to explicitly call queue runners. But apparently there is no support for truncated backpropagation through time when I use this API.

Tensorflow Version: 1.11.0

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by ZeRoGerc** [comment]
In the meanwhile. It is okay to run session to get the value parsed with `tf.data` and then split it and use `batch_sequences_with_states` for TBPTT later like this:

```
batch = iterator.get_next()
numpy_batch = sess.run(batch)
train_with_batch_sequences_with_states(numpy_batch)
```

Or is there are a better approach? I don't want to rewrite my whole pipeline build with tf.data just for TBPTT.

Thanks in advance.

**Comment by rsepassi** [comment]
Adding @mrry to route to the right person within `tf.data`

**Comment by mrry** [comment]
Thanks Ryan. Passing this over to @ebrevdo, who added some `tf.data`-based training utilities that could be helpful here.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ebrevdo: It has been 17 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @ebrevdo: It has been 31 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by HansikaPH** [comment]
Thanks for the response @ebrevdo. I have tried doing exactly the thing that you have mentioned. But the process seems to be quite slow. That is why I wanted to know if there is some internally optimised method of doing that.

**Comment by rmothukuru** [comment]
@ghost,
Sorry for the delayed response. **`RNNs`** now support **`tf.data`**. Please refer [this code in the documentation](https://www.tensorflow.org/text/tutorials/text_generation#create_training_batches) and let us know if this is what you are looking for. Thanks!
